File: packages/java/src/main/java/com/ummalqura/DayOfWeek.java
package com.ummalqura;

/**
 * Day of week information
 */
public class DayOfWeek {
    private final int index;
    private final String nameEn;
    private final String nameAr;

    public DayOfWeek(int index, String nameEn, String nameAr) {
        this.index = index;
        this.nameEn = nameEn;
        this.nameAr = nameAr;
    }

    public int getIndex() { return index; }
    public String getNameEn() { return nameEn; }
    public String getNameAr() { return nameAr; }
}

File: packages/php/src/DayOfWeek.php
<?php

namespace UmmAlQura;

class DayOfWeek
{
    public function __construct(
        private int $index,
        private string $nameEn,
        private string $nameAr
    ) {}

    public function getIndex(): int { return $this->index; }
    public function getNameEn(): string { return $this->nameEn; }
    public function getNameAr(): string { return $this->nameAr; }

    public function toArray(): array
    {
        return [
            'index' => $this->index,
            'name_en' => $this->nameEn,
            'name_ar' => $this->nameAr,
        ];
    }
}
